### ISSUE TITLE
drop Python 2 support and new release

### ISSUE THREAD

**Issue by otizonaizit** [mdp-toolkit/mdp-toolkit#55]
Python 2 will not be maintained anymore starting from 2020. Given that the last MDP release is more than 3 years old, I think it's time to make a new release with the following focus:

  - drop Python 2 compatibility
  - finish whatever needs to be finished

I volunteer to take care of the release, but you guyes have to merge the current development branches into master, so that I know **what** to release.

The release is urgent also because Fedore and Debian/Ubuntu are dropping Python 2 and we need a new MDP package or we will be kicked out of their repos.
What do you think?

**Comment by Stewori** [comment]
I am on vacation at the moment.

**Comment by Stewori** [comment]
I agree on dropping official Python 2 support. Given that the code is currently de-facto Python 2 compatible I'd like to keep it that way for this release. Also, having a designated last release that runs  on Python 2 is imo best practice. I think it will stay in use for some more years. Dropping Python 2 as official dependency and for the package is the ideal compromise.

**Comment by otizonaizit** [comment]
OK, so the plan is to keep Python 2 support for this release, and drop it from after the release on.

But is there any "content" work in the pipeline, or can I just cut a release right now? There were some PRs and commits in the last months: is everything merged from there?

Note that I won't have time to update the website and the docs. I can take care of testing that the everything works and make a release on GitHub and of the Debian/Ubuntu package.

**Comment by Stewori** [comment]
https://github.com/mdp-toolkit/mdp-toolkit/pull/48 should get in. I think it has still some doc issues, e.g. the doc misconcepts training in chunks vs phases.

I am traveling until early December and would like to give it another review before merging.

The other open pr won't go in. It breaks Python 3.5 compatibility without compelling benefit.

**Comment by Stewori** [comment]
Green light for the release from me. Please let me know if we can help with any step of the process.

**Comment by Stewori** [comment]
Had a look at https://github.com/mdp-toolkit/mdp-toolkit/blob/master/CHECKLIST and a couple of things appear outdated. Note that one has to pin `'pytest==4.0.0'` for the pip commands at the moment (see #66). It would be good if the CHECKLIST could be updated to the current situation, most notably SF->GH.

**Comment by Stewori** [comment]
In CHECKLIST, the pypi release process looks rather complicated to me. Maybe we can switch to an automated process...? Nowadays, travis takes over the testing part. In [pytypes](https://github.com/Stewori/pytypes) I added a deploy section to travis.yml, see https://github.com/Stewori/pytypes/blob/master/.travis.yml. This automatically forwards github releases to pypi. All I have to do there for a new release is to file it on github.
@otizonaizit that would require you to provide a public key for your pypi password like here: https://github.com/Stewori/pytypes/blob/master/.travis.yml#L23
I do not remember how I generated it but I'm sure we can figure it out if you like this proposal.

Sure, this does not consider the documentation but that's a different story anyway. For now, I would disentangle documentation releases from mdp releases. (Until we know where the documentation is going...)

And finally, I don't have a clue about the debian package release process as CHECKLIST does not seem to cover it.
